sled-agent/src/services.rs

```diff
@@ -2908,45 +2973,62 @@ impl ServiceManager {
+                                    "interfaces",
+                                    "astring",
+                                    &i.to_string(),
+                                );
+                            }
 
                             if is_gimlet {
-                                // Ddm for a scrimlet needs to be configured to
-                                // talk to dendrite
-                                smfh.setprop("config/dpd_host", "[::1]")?;
-                                smfh.setprop("config/dpd_port", DENDRITE_PORT)?;
+                                mg_ddm_config = mg_ddm_config
+                                    .add_property(
+                                        "dpd_host", "astring", "[::1]",
+                                    )
+                                    .add_property(
+                                        "dpd_port",
+                                        "astring",
+                                        &DENDRITE_PORT.to_string(),
+                                    )
                             }
-                            smfh.setprop("config/dendrite", "true")?;
 
-                            smfh.refresh()?;
+                            mg_ddm_service = mg_ddm_service.add_instance(
+                                ServiceInstanceBuilder::new("default")
+                                    .add_property_group(mg_ddm_config),
+                            );
                         }
                     }
-
-                    debug!(self.inner.log, "enabling service");
-                    smfh.enable()?;
                 }
-            }
-        };
 
-        Ok(running_zone)
+                switch_zone_setup_service = switch_zone_setup_service
+                    .add_instance(
+                        ServiceInstanceBuilder::new("default")
+                            .add_property_group(switch_zone_setup_config),
+                    );
+
+                let profile = ProfileBuilder::new("omicron")
+                    .add_service(nw_setup_service)
+                    .add_service(disabled_dns_client_service)
+                    .add_service(mgs_service)
+                    .add_service(wicketd_service)
+                    .add_service(switch_zone_setup_service)
+                    .add_service(dendrite_service)
+                    .add_service(tfport_service)
+                    .add_service(lldpd_service)
+                    .add_service(pumpkind_service)
+                    .add_service(mgd_service)
+                    .add_service(mg_ddm_service)
+                    .add_service(uplink_service);
+                profile
+                    .add_to_zone(&self.inner.log, &installed_zone)
+                    .await
+                    .map_err(|err| {
+                        Error::io("Failed to setup Switch zone profile", err)
+                    })?;
+                return Ok(RunningZone::boot(installed_zone).await?);
```

Comment:
Sweeeeeeeet, love to see this -- booting being the last step is perfect, this is absolutely what I was hoping for (and I wasn't even 100% sure if it was possible)!